src/core/dynamic-user.c

```diff
@@ -223,6 +259,7 @@ static int pick_uid(const char *name, uid_t *ret_uid) {
                 }
 
                 (void) ftruncate(lock_fd, l);
+                (void) make_uid_symlinks(candidate, name, true); /* also add direct lookup symlinks */
```

Comment:
Failure is silent here. We don't expect this to fail, but if it does, it would result in hard to diagnose failure (dbus communication errors). I think the two symlink() calls in make_uid_symlinks() should result in log_warning on failure.